breakdown cli quickstart steps

mirnawong1 opened the pull request:
the core quickstart currently embeds multiple h3 steps into one h2. this makes the pages quite long and misrepresents the steps as well. 

After discussion with @runleonarun and feedback from @greg-mckeon , this pr breaks down the h3 steps into h2 to give users more bite-sized chunks of content.

this pr also adds redirects from steps to anchored links.

[docs project](https://www.notion.so/dbtlabs/Breakdown-Core-quickstart-into-steps-0f1d500635fa4d288371dc4a033cc4e4)

Commit:
split h3s into h2s to make quickstarts more clear

Commit:
add image

Commit:
change to h2

mirnawong1 committed:
Merge branch 'current' into mwong-breakdown-core-qs

runleonarun committed:
fixing old bug

runleonarun committed:
revising some spacign issues that have beenthere forever

runleonarun committed:
one mor espacign issue

runleonarun commented on website/static/_redirects:
```diff
@@ -1,3 +1,11 @@
+
+# Redirects for quickstarts sections that were split
+
+/quickstarts/manual-install?step=2 /quickstarts/manual-install?step=2 301
```
@mirnawong1 I'm going to remove the redirects so it won't affect future links to the correct steps. Also, not sure we can do redirects at this level
```suggestion
```

mirnawong1 replied:
awesome! should we do a crawler trigger after this pr is merged @runleonarun ? just in case

runleonarun replied:
@mirnawong1 probably a good idea. I can try out our new process!

runleonarun commented on website/static/_redirects:
```diff
@@ -1,3 +1,11 @@
+# Redirects for quickstarts sections that were split
+
+/quickstarts/manual-install?step=2 /quickstarts/manual-install?step=2 301
+/quickstarts/manual-install?step=3 /quickstarts/manual-install?step=7 301
```
```suggestion
```

runleonarun commented on website/static/_redirects:
```diff
@@ -1,3 +1,11 @@
+
+/quickstarts/manual-install?step=2 /quickstarts/manual-install?step=2 301
+/quickstarts/manual-install?step=3 /quickstarts/manual-install?step=7 301
+/quickstarts/manual-install?step=4 /quickstarts/manual-install?step=13 301
```
```suggestion
```

runleonarun commented on website/static/_redirects:
```diff
@@ -1,3 +1,11 @@
+/quickstarts/manual-install?step=2 /quickstarts/manual-install?step=2 301
+/quickstarts/manual-install?step=3 /quickstarts/manual-install?step=7 301
+/quickstarts/manual-install?step=4 /quickstarts/manual-install?step=13 301
+/quickstarts/manual-install?step=5 /quickstarts/manual-install?step=15 301
```
```suggestion
```

runleonarun committed:
Apply suggestions from code review

runleonarun commented on website/static/_redirects:
```diff
@@ -1,3 +1,7 @@
+
+# Redirects for quickstarts sections that were split
```
```suggestion
```

runleonarun committed:
Update website/static/_redirects

runleonarun commented on website/static/_redirects:
```diff
@@ -1,3 +1,6 @@
+
```
```suggestion
```

runleonarun commented on website/static/_redirects:
```diff
@@ -1,3 +1,6 @@
+
+
```
```suggestion
```

runleonarun commented on website/static/_redirects:
```diff
@@ -1,3 +1,6 @@
+
+
+
```
```suggestion
```

runleonarun committed:
Apply suggestions from code review

runleonarun committed:
one mor espacign issue

runleonarun committed:
Merge branch 'mwong-breakdown-core-qs' of github.com:dbt-labs/docs.getdbt.com into mwong-breakdown-core-qs

runleonarun reviewed (approved):
Looks good once John helps me figure out why I'm getting an error when I refresh the page.

john-rock committed:
Refactor Quickstart component to prevent error (#4030)

## What are you changing in this pull request and why?

Fixes the `Node.removeChild: The node to be removed is not a child of
this node` error we were seeing when trying to use a Snippet with a
nested <WHcode /> component.

## Additional information

The following code was setup to make sure that in the event a snippet
that is nested under a step (##) in the quickstart markdown file
contained an h2 (##) it would not be rendered as an additional step.
```
      snippetContainer.forEach((snippet) => {
        const parent = snippet?.parentNode;
        while (snippet?.firstChild && parent.className) {
          if (parent) {
            parent.insertBefore(snippet.firstChild, snippet);
          }
        }
      });
```

Since the update core quickstart contains a snippet in Step 8 `<Snippet
path="tutorial-sql-query" />` which contains a nested component `<WHcode
/>` it was causing the component to break when building the steps.

This PR makes sure the component is fully loaded before attempting any
manipulation on the steps.

runleonarun committed:
Merge branch 'current' into mwong-breakdown-core-qs

runleonarun committed:
Merge branch 'current' into mwong-breakdown-core-qs
